src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -2137,7 +2136,7 @@ _public_ int sd_varlink_observe(sd_varlink *v, const char *method, sd_json_varia
         r = sd_json_buildo(
                         &m,
                         SD_JSON_BUILD_PAIR("method", SD_JSON_BUILD_STRING(method)),
-                        SD_JSON_BUILD_PAIR("parameters", SD_JSON_BUILD_VARIANT(parameters)),
+                        SD_JSON_BUILD_PAIR_CONDITION(!sd_json_variant_is_blank_object(parameters), "parameters", SD_JSON_BUILD_VARIANT(parameters)),
```

Comment:
hmm, so we have the internal helper macro JSON_BUILD_PAIR_VARIANT_NON_NULL() already, and this is very close to that. Maybe add JSON_BUILD_PAIR_VARIANT_NON_EMPY() or so in the same style, which makes this more readable and shorter.